### DISCUSSION TITLE
Unable to authenticate airflow CLI following the Google OpenID Airflow doc

### DISCUSSION THREAD

**Discussion by sumeshpremraj** [apache/airflow#32021] [General]
I want to run dags backfill (and later document this for end users of Airflow), by connecting a local instance of airflow CLI to the remote Airflow server which runs tasks. Airflow runs on GCP, and auth is via Google OpenID. Our end users are also all Airflow admins.

I am following this [Google OpenID](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/api-auth-backend/google-openid.html) doc, and am unable to get past the `gcloud auth print-identity-token` command.

What should be the value for --audience parameter for this command?
In [gcloud doc](https://cloud.google.com/sdk/gcloud/reference/auth/print-identity-token#--audiences), I see it should be the protected resource URL, so I tried using the URL of my airflow instance. 

I get `Invalid account Type for --audiences. Requires valid service account` error. What am I doing wrong?

Commands used:
```
$ ENDPOINT_URL="https://[airflow URL]"
$ AUDIENCE="[airflow URL]"
$ ID_TOKEN="$(gcloud auth print-identity-token "--audiences=${AUDIENCE}")"
ERROR: (gcloud.auth.print-identity-token) Invalid account Type for `--audiences`. Requires valid service account.
```

If I misunderstood the documentation, I would like to understand it better, and then I can contribute improvements.

**Comment by potiuk** [comment]
I thikn it's question to gcloud team . Not sure why you are asking here.

**Reply by sumeshpremraj** [reply]
This is a question about the correctness and clarity of https://airflow.apache.org/docs/apache-airflow-providers-google/stable/api-auth-backend/google-openid.html

**Comment by potiuk** [comment]
~BTW In this particular case apparently it's the 'audiences' parameter that is wrong - and it comes from  gcloud - so likely you need to find out there~

**Comment by potiuk** [comment]
Actually it seems you have no valid service Account (so the message is right). If you look few lines above you see the 'default application credentials'. My wild guess is that you need to setup service Account this way (so for example point to the key of yours with the right env variable). 

I think there is no way for the gcloud command to read airflow configurationn. So if you only configured key in airflow configuration but not in default credentials, gcloud command needs to get the credentials from somewhere and there is no physical possibility it takes it from airflow configuration

So you need to figure out how to pass service Account to the gcloud command. It might be likely misleading but this is logical explanation when you look at this. 

Once you figure out how to do it best - update to the docs 'suggest a change on this page' are most welcome

**Reply by potiuk** [reply]
Note - i am just guessing - not the author of the openid docs, it just comes from generic computing knowledge and guessing what could be wrong - the usual debugging. simply runnig gcloud command  without passing it any credentials, means that default credentials will be used. If you do not have them - the gcloud command will not be able to authenticate. That's logical explanation.

**Reply by potiuk** [reply]
do you have default credentials set up ? I am curious if my guess is good.
